lib/apiservers/engine/backends/eventmonitor.go

```diff
@@ -236,8 +238,21 @@ func (p DockerEventPublisher) PublishEvent(event plevents.BaseEvent) {
 			attrs["exitCode"] = code
 			log.Infof("Sending die event for container %s - code: %s", vc.ContainerID, code)
 			actor := CreateContainerEventActorWithAttributes(vc, attrs)
-
 			EventService().Log(containerDieEvent, eventtypes.ContainerEventType, actor)
+
+			// auto-remove if required
+			if vc.HostConfig.AutoRemove {
```

Comment:
My inclination would be to record the autoremove status in container annotations along with the other docker specific info tightly associated with a single container.
If we need auto-remove to support 1.13 client sanely, then we can merge this and open an issue to persist the value.
If we do not need auto-remove then I'm inclined to fix this first so we don't have unpredictable behaviour.